command/server/config.go

```diff
@@ -828,6 +852,30 @@ func parseHAStorage(result *Config, list *ast.ObjectList, name string) error {
 	return nil
 }
 
+func parseServiceDiscovery(result *Config, list *ast.ObjectList, name string) error {
+	if len(list.Items) > 1 {
+		return fmt.Errorf("only one %q block is permitted", name)
+	}
+
+	// Get our item
+	item := list.Items[0]
+	key := name
+	if len(item.Keys) > 0 {
+		key = item.Keys[0].Token.Value().(string)
+	}
+
+	var m map[string]string
```

Comment:
done

Reply:
I reverted this change.  If we do this, we have to change the config for consul too, which end up being a pretty big change since you have to cast all the map values to a string.  Lets discuss and see if we want to do this.

Reply:
I'll go ahead and resolve this for now.  We can revisit the idea later